tests/vcs/VCSCommitBearTest.py

```diff
@@ -0,0 +1,181 @@
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of the message strings.
+        """
+        # return list(self.uut.run(*args, **kwargs))
+        return list(result.message for result in self.uut.run(*args, **kwargs))
+
+    def assert_no_msgs(self):
+        """
+        Assert that there are no messages in the message queue of the bear, and
+        show the messages in the failure messgae if it is not empty.
+        """
+        self.assertTrue(
+            self.msg_queue.empty(),
+            'Expected no messages in bear message queue, but got: ' +
+            str(list(str(i) for i in self.msg_queue.queue)))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = GitCommitBear(None, self.section, self.msg_queue)
+
+        self._old_cwd = os.getcwd()
+        self.gitdir = mkdtemp()
+        os.chdir(self.gitdir)
+        self.run_git_command('init')
+        self.run_git_command('config', 'user.email coala@coala.io')
+        self.run_git_command('config', 'user.name coala')
+
+    @staticmethod
+    def _windows_rmtree_remove_readonly(func, path, excinfo):
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.gitdir, onerror=onerror)
+
+    def test_analyze_git_commit(self):
+        run_shell_command('touch "abc" >> testfile1.txt ')
+        run_shell_command('git add testfile1.txt')
+        run_shell_command('git commit -m "Add testfile1"')
+        self.assertEqual(self.run_uut(), [])
+
+        run_shell_command('touch "abcdabcd" >> testfile2.txt ')
+        run_shell_command('git add testfile2.txt')
+        run_shell_command('git commit -m "Add testfile2 [ci skip]"')
+        self.assertEqual(self.run_uut(), [])
+
+        run_shell_command('touch "abcdefabcdef" >> testfile3.txt ')
+        self.run_git_command('add', 'testfile3.txt')
+        self.run_git_command('commit', '-m', '"another commit [skip ci]"')
+        self.run_git_command('revert', 'HEAD', '--no-edit')
+        self.assertEqual(self.run_uut(), [])
+
+        run_shell_command('git checkout -b new-feature master')
+        run_shell_command('touch "some text" >> testfile4.txt ')
+        self.run_git_command('add', 'testfile4.txt')
+        self.run_git_command('commit', '-m', '"Commit in feature branch"')
+        run_shell_command('git checkout master')
+        run_shell_command('git merge new-feature')
```

Comment:
Again wrt https://gitter.im/coala/2018-gsoc-all?at=5b2a6f0dad21887018cadd01
This is not even creating a merge commit, it is simply creating a normal commit which the `git merge` will simply fast forward and there will be *only one* parent commit.